### PR TITLE
Oe/86bz12tn4/realization stepper v2

### PR DESCRIPTION
**Pull request by guardian-001**
I ran this PR just to ensure that the number of files changed is not high. The work is not complete
folders and files structure will be updated in the last pr

### CODE REVIEW

File: src/api/architect/project/use-needs.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import { client } from '@/api/common';
+import type { Need } from '@/types';
+
+type Response = Need[];
+type Variables = void;
+
+export const useNeedsApi = createQuery<Response, Variables, AxiosError>({
+  queryKey: ['categories'],
+  fetcher: () => {
```
Comment (Ghazi-Chaftar) [inline src/api/architect/project/use-needs.tsx:12]: extract into a service file

File: src/api/architect/user/use-data-architect.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// api/styles.ts
```
Comment (Ghazi-Chaftar) [inline src/api/architect/user/use-data-architect.tsx:1]: remove gpt comment

File: src/app/(supplier)/(public)/login.tsx [hunk 3]
```diff
@@ -13,11 +12,14 @@ export type LoginFormSupplierProps = {
 };
 export default function Login() {
   const router = useRouter();
-  const signIn = useAuth.use.signIn();
+  // const signIn = useAuth.use.signIn();
```
Comment (Ghazi-Chaftar) [inline src/app/(supplier)/(public)/login.tsx:15]: comment

File: src/app/(supplier)/(public)/login.tsx [hunk 4]
```diff
@@ -13,11 +12,14 @@ export type LoginFormSupplierProps = {
...
   useSoftKeyboardEffect();
 
   const onSubmit: LoginFormSupplierProps['onSubmit'] = () => {
-    signIn({ access: 'access-token', refresh: 'refresh-token' });
+    // signIn({
```
Comment (Ghazi-Chaftar) [inline src/app/(supplier)/(public)/login.tsx:19]: comment

File: src/modules/architect/realization/shared/components/image-picker.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      <Text
+        tx={'realisation.galleryStep.imgSelectorPlaceholder'}
+        className="text-md max-w-xs text-center text-description"
+      />
+      <StepperButton
+        width="w-[45%]"
+        alternativeBg="bg-secondary-btn"
+        alternativeTextStyle="color-primary-txt"
+        label={translate('realisation.galleryStep.imgSelectorPlaceholder')}
+        onPressHandler={() => {}} //uploadImages
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/realization/shared/components/image-picker.tsx:23]: empty function ! remove it please

Comment (Ghazi-Chaftar) [inline src/modules/architect/realization/shared/components/image-picker.tsx:23]: remove comment

File: src/modules/architect/realization/steps/adopted-style.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    ArchitecturalStyleSchema,
+    {
+      architecturalStyle: formData.architecturalStyle,
+    }
+  );
+
+  const { data, isPending, isError } = useStylesApi();
+
+  const onSubmit = (selectedData: CategoryFormType) => {
+    setFormData((prev: any) => ({
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/realization/steps/adopted-style.tsx:29]: can't we type this

File: src/modules/architect/realization/steps/approved-services.tsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+import type { ProjectRealizationType } from '../types';
+
+export function ApprovedServices() {
+  type NeedsFormType = Pick<ProjectRealizationType, 'needs'>;
+  const { formData, setFormData, onHandleNext, onHandleBack } =
+    useFormStepper<ProjectRealizationType>();
+  const {
+    handleSubmit,
+    control,
+    // errors
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/realization/steps/approved-services.tsx:21]: remove comment

File: src/modules/architect/realization/steps/approved-services.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+  } = useCustomForm(NeedsSchema, {
+    needs: formData.needs,
+  });
+  console.log(getUser());
+  const { data, isPending, isError, error } = useNeedsApi();
+  console.log('error: ', error);
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/realization/steps/approved-services.tsx:27]: remove console log

File: src/modules/architect/realization/steps/approved-services.tsx [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+  const onSubmit = (selectedData: NeedsFormType) => {
+    setFormData((prev: any) => ({
+      ...prev,
+      ...selectedData,
+    }));
+    onHandleNext();
+  };
+
+  // const error = errors.needs?.message as TxKeyPath | undefined;
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/realization/steps/approved-services.tsx:37]: comment

File: src/modules/architect/realization/steps/approved-services.tsx [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+  return (
+    <View className="mb-5 flex h-full w-full flex-1 items-start justify-between gap-6  ">
+      {isError ? (
+        <ErrorData message="Error Loading Data" />
+      ) : (
+        <>
+          {isPending ? (
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/realization/steps/approved-services.tsx:44]: dont use ? : use && instead for more readablity of the code

File: src/modules/architect/realization/steps/project-details.tsx [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+    ProjectDetailsSchema,
+    {
+      projectName: formData?.projectName,
+      city: formData?.city,
+      workSurface: formData?.workSurface,
+      description: formData?.description,
+    }
+  );
+
+  useEffect(() => {
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/realization/steps/project-details.tsx:38]: remove this

File: src/modules/architect/realization/steps/project-details.tsx [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+  });
+  const onSubmit = (data: DetailsFormType) => {
+    console.log(data);
+    setFormData((prev: any) => ({
+      ...prev,
+      ...data,
+    }));
+    onHandleNext();
+  };
+  // const projectNameError = errors.description?.message as TxKeyPath | undefined;
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/realization/steps/project-details.tsx:49]: comment

File: src/modules/login/login-form.tsx [hunk 13]
```diff
@@ -31,7 +31,11 @@ export const LoginForm = () => {
   const onSubmit = (data: LoginFormType) => {
     login.mutate(data, {
       onSuccess: (response) => {
-        signIn({ access: response.access, refresh: response.refresh });
+        console.log(response);
```
Comment (Ghazi-Chaftar) [inline src/modules/login/login-form.tsx:34]: console log

File: src/shared/components/image-uploader.tsx [hunk 14]
```diff
@@ -0,0 +1,113 @@
+// import React from 'react';
```
Comment (Ghazi-Chaftar) [inline src/shared/components/image-uploader.tsx:1]: remove this file

**Review by Ghazi-Chaftar — CHANGES_REQUESTED** [review]
left comments

File: src/modules/architect/realization/steps/approved-services.tsx (hunk 10 above)
Comment (guardian-001) [reply src/modules/architect/realization/steps/approved-services.tsx:44]: i need the else property here, it's better to use ?

File: src/modules/architect/realization/steps/approved-services.tsx (hunk 9 above)
Comment (guardian-001) [reply src/modules/architect/realization/steps/approved-services.tsx:37]: removed

File: src/modules/architect/realization/steps/approved-services.tsx (hunk 8 above)
Comment (guardian-001) [reply src/modules/architect/realization/steps/approved-services.tsx:27]: removed

File: src/modules/architect/realization/steps/approved-services.tsx (hunk 7 above)
Comment (guardian-001) [reply src/modules/architect/realization/steps/approved-services.tsx:21]: removed

File: src/modules/architect/realization/steps/adopted-style.tsx (hunk 6 above)
Comment (guardian-001) [reply src/modules/architect/realization/steps/adopted-style.tsx:29]: already in last pr

File: src/modules/architect/realization/shared/components/image-picker.tsx (hunk 5 above)
Comment (guardian-001) [reply src/modules/architect/realization/shared/components/image-picker.tsx:23]: removed

Comment (guardian-001) [reply src/modules/architect/realization/shared/components/image-picker.tsx:23]: i already added a comment to explain that this function will be in the future a function to upload images, 
already changed in last pr

File: src/app/(supplier)/(public)/login.tsx (hunk 4 above)
Comment (guardian-001) [reply src/app/(supplier)/(public)/login.tsx:19]: removed

File: src/app/(supplier)/(public)/login.tsx (hunk 3 above)
Comment (guardian-001) [reply src/app/(supplier)/(public)/login.tsx:15]: removed

File: src/api/architect/user/use-data-architect.tsx (hunk 2 above)
Comment (guardian-001) [reply src/api/architect/user/use-data-architect.tsx:1]: removed

File: src/modules/architect/realization/steps/project-details.tsx (hunk 11 above)
Comment (guardian-001) [reply src/modules/architect/realization/steps/project-details.tsx:38]: removed

File: src/modules/architect/realization/steps/project-details.tsx (hunk 12 above)
Comment (guardian-001) [reply src/modules/architect/realization/steps/project-details.tsx:49]: removed

File: src/modules/login/login-form.tsx (hunk 13 above)
Comment (guardian-001) [reply src/modules/login/login-form.tsx:34]: removed

File: src/shared/components/image-uploader.tsx (hunk 14 above)
Comment (guardian-001) [reply src/shared/components/image-uploader.tsx:1]: removed

**Pull request merged**